tex/polyglossia.sty

```diff
@@ -1358,25 +1367,40 @@
 }
 
 \prg_set_conditional:Npnn \polyglossia@check@ifdefined:N #1 { p , T , F , TF }{
-  \cs_if_exist:cTF{l@#1}{
-    \cs_if_eq:cNTF{l@#1}{\l@nohyphenation}{\prg_return_false:}{\prg_return_true:}
-  }{
-    \prg_return_false:
-  }
+  \cs_if_exist:cTF {l@#1}
+    {
+      \cs_if_eq:cNTF {l@#1} \l@nohyphenation
+        {
+          \prg_return_false:
+        }
+        {
+          % it's possible that sometimes \csname l@#1\endcsname becomes \relax
+          \cs_if_eq:cNTF {l@#1} \relax
+            { \prg_return_false: }
+            { \prg_return_true: }
+        }
+    }
+    {
+      \prg_return_false:
+    }
+}
+
+\def\polyglossia@luatex@load@lang#1{%
+  % if \l@#1 is not properly defined, call lua function newloader(#1),
+  % and assign the returned number to \l@#1
+  \polyglossia@check@ifdefined:NF {#1}
+    {
+      \expandafter\chardef\csname l@#1\endcsname=
+        \directlua{ tex.sprint(polyglossia.newloader'#1') }\relax
+    }
 }
 
 \newcommand\xpg@ifdefined[3]{%
-    % With luatex, we first need to define \l@##1. We only do that if the pattern exists.
-    \ifluatex%
-       \ifstrequal{#1}{latex}{}{%
```

Comment:
This check for latex is mandatory. Otherwise, we will get complains about the latex language (which is only used internally)

Reply:
No, we won't get complains. After loading of polyglossia package, `\l@latex` is already defined as `\char 0`. And the macros `\polyglossia@luatex@load@lang` and `\polyglossia@check@ifdefined:NF` will check if `\l@latex` is defined; as it is already defined, `newloader('latex')` will never be triggered.

Reply:
OK, if you say so. I have added that check because there was such an issue, but maybe it is resolved with your redef.